packages/main/src/TimePickerClock.ts

```diff
@@ -0,0 +1,991 @@
+	 *
+	 * - on both device types, if there is a keyboard attached: 24 or 00 can be typed directly.
+	 *
+	 * <b>Note:</b> Don't use it together with am/pm.
+	 *
+	 * @name sap.ui.webc.main.TimePickerClock.prototype.support2400
+	 * @type {boolean}
+	 * @defaultvalue false
+	 */
+	@property({ type: Boolean })
+	support2400!: boolean;
+
+	/**
+	 * Defines the currently available Time Picker Clock items depending on Clock setup.
+	 *
+	 * @type {Array}
+	 */
+	@property({ type: Object, multiple: true })
+	_items!: Array<TimePickerClockItem>;
+
+	/**
+	 * Defines the currently selected Time Picker Clock item.
+	 *
+	 * @type {TimePickerClockSelectedItem}
+	 */
+	@property({ type: Object })
+	_selectedItem!: TimePickerClockSelectedItem;
+
+	/**
+	 * Keeps variables used in interaction calculations.
+	 *
+	 * @type {TimePickerClockDimensions}
+	 */
+	@property({ type: Object })
+	_dimensionParameters!: TimePickerClockDimensions;
+
+	/**
+	 * Mousedown or Touchstart event flag.
+	 *
+	 * @type {boolean}
+	 * @defaultvalue false
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_mouseOrTouchDown!: boolean;
+
+	/**
+	 * Cancel Mouseout flag.
+	 *
+	 * @type {boolean}
+	 * @defaultvalue false
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_cancelTouchOut!: boolean;
+
+	/**
+	 * Visibility of '24' on Hours clock flag.
+	 *
+	 * @type {boolean}
+	 * @defaultvalue false
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_is24HoursVisible!: boolean;
+
+	/**
+	 * Calculated selected value of the clock during interactions.
+	 *
+	 * @type {integer}
+	 * @defaultvalue -1
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_selectedValue!: number;
+
+	/**
+	 * Selected value of the clock during interactions.
+	 *
+	 * @type {integer}
+	 * @defaultvalue -1
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_movSelectedValue!: number;
+
+	/**
+	 * Hovered value of the clock during interactions.
+	 *
+	 * @type {integer}
+	 * @defaultvalue -1
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_hoveredValue!: number;
+
+	/**
+	 * Previously hovered value of the clock during interactions.
+	 *
+	 * @type {integer}
+	 * @defaultvalue -1
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_prevHoveredValue!: number;
+
+	/**
+	 * Animation in progress flag.
+	 *
+	 * @type {boolean}
+	 * @defaultvalue false
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_animationInProgress!: boolean;
+
+	/**
+	 * Stores the ID of the long touch timeout.
+	 *
+	 * @type {integer}
+	 * @defaultvalue -1
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_longTouchId!: number;
+
+	constructor() {
+		super();
+		// attach global event
+		document.addEventListener("mouseup", this._onMouseOutUp.bind(this), false);
```

Comment:
It's a good practice to remove the listener
```js
// See an example in the ListItem.js
onEnterDOM() {
	document.addEventListener("mouseup", this.deactivate););
}

onExitDOM() {
	document.removeEventListener("mouseup", this.deactivate);
}
```

Reply:
Done.